### PR TITLE
Add partial support for French prompt sets

### PR DESCRIPTION
**Pull request by rogthefrog**
This PR contains a kludge we should resolve better as a group.

tl;dr single-locale (or single-hazard) prompt set files are harder to add than they should be because of how we reference prompt set names and prompt set file names, AND the way the dependency helper validates a SUT has the right dependencies and those dependencies are cached.

tl;dr hack here: 

* you must append the locale string to the prompt set file name when you upload it (e.g. for French, name your upload `airr_official_1.0_practice_prompt_set_release_fr_fr.csv`)
* delete any prompt set file you have locally in `./run/tests/<prompt set name>` before running a benchmark

# Observations

Prompt set names and ostensible file names in safe_v1 are 
```
PROMPT_SETS = {
    "practice": "airr_official_1.0_practice_prompt_set_release",
    "official": "airr_official_1.0_heldback_prompt_set_release",
}
```
1. The **keys** (or **nicknames**) ("practice" and "official") are hard-coded in modellab and modelrunner, so it's non-trivial to add another for things like "the practice prompt set, but just this locale in it" and have that option available everywhere.
2. The **names** (`airr_official_1.0_(practice|heldback)_prompt_set_release`) are used as a) the dependency helper cache directory (`./run/tests/airr_official_1.0_(practice|heldback)_prompt_set_release`), b) the dict key used to check if the SUT has all its dependencies, and c) the actual **filename** on the server when the file is uploaded to modellab.
3. The safe test knows how to handle multiple locales in a prompt set file, so it's easy to say "use this prompt set with all the languages in it, but just the French locale," but because of point (2) it's harder to do the reverse, i.e. have multiple files for a single "prompt set" (e.g. official 1.0 french, official 1.0 english, etc.).

## Specific Issue That Led To This

* have the English `airr_official_1.0_practice_prompt_set_release.csv` file previously downloaded locally by a run in `run/tests/airr_official_1.0_practice_prompt_set_release/somehash`
* try to run a benchmark using this code with the French locale, which will set `airr_official_1.0_practice_prompt_set_release_fr_fr.csv` (the filename) as a dependency keyed by `airr_official_1.0_practice_prompt_set_release` (the prompt set name). The dependency helper will think it's already there and try to find test items in French in a file that only contains prompts in English. Spoiler: this ends poorly.

Deleting the cached version makes this code work, but it's not a good solution.

# Solution in This PR

* The safe test dependency helper appends the locale string to the prompt set name to generate the filename to  download, if the locale isn't all, blank, or English. 
  * E.g. the **prompt set name** "airr_official_1.0_practice_prompt_set_release" becomes the "airr_official_1.0_practice_prompt_set_release_fr_fr.csv" **filename**.
* You need to delete any preexisting cached version of that same prompt set if it is in a different locale, because the dependency helper will download them all into the same cache dir.
* I reuploaded Kurt's test file with `_fr_fr` in a different spot in the **filename**.

# Ideas

* Decouple prompt set **keys** (or nicknames, e.g. "official" and "practice") from **filenames** in modellab, from dependency caching directory names.
* Make it easier to share identifiers across the apps (modelbench CLI, docker entrypoint command, modelrunner, modellab).
* Make it easier for the operator (or for modelbench) to know what prompt set files are available; maybe publish the list and enumerate them in the CLI to guide the operator?

### CODE REVIEW

**Review by bkorycki — APPROVED** [review]
looks good to me!

File: src/modelgauge/tests/safe_v1.py [hunk 1]
```diff
@@ -131,15 +138,25 @@ def _check_annotators(cls):
...
     def get_dependencies(self) -> Mapping[str, ExternalData]:
-        modellab_base_download_url = "https://modellab.modelmodel.org/files/download"
+        prompt_set_base_download_url = f"https://{PROMPT_SET_DOWNLOAD_HOST}/files/download"
         headers = None
         if self.token is not None:
             headers = {"auth-token": self.token.value}
-        # Only one dependency.
+        source_url = f"{prompt_set_base_download_url}/{self.__localize_filename()}.csv"
         return {
             self.prompt_set_file_name: WebData(
```
[the remark this answers is not in the document]
Comment (rogthefrog) [reply src/modelgauge/tests/safe_v1.py:158]: It currently doesn't include the locale, because that string is also the "family" of prompts (the set of prompts from a specific vendor at a specific version, with a practice vs. official flag). The prompt file can contain multiple locales. The file name string is a value in a dict key with two keys ("official" and "practice") that are used in other apps as well (modellab and modelrunner). Having multiple files (e.g. the same vendor/version of prompts but different locales) with distinct names would require changing modellab and modelrunner too, and the way dependencies are managed. I tried that approach and it got ugly quickly. There's more detail in the PR description.

File: src/modelbench/run.py [hunk 2]
```diff
@@ -98,7 +99,7 @@ def cli() -> None:
 @click.option(
     "--locale",
     "-l",
-    type=click.Choice(["en_us"], case_sensitive=False),
+    type=click.Choice(["en_us", "fr_fr"], case_sensitive=False),
```
[the remark this answers is not in the document]
Comment (rogthefrog) [reply src/modelbench/run.py:102]: Great idea.
